### PR TITLE
Obg 440

### PR DESCRIPTION
**Pull request by electronicpeter**
1 removed all requestSession stuff from Roman, as it is no more used (idea was to repeat call)
  2 removed xsrf token from persistence
  3 fixed bug: after logout consent had to be retrieved again.
  4 simulated call 4c) e.g. as long as redirectOK is not call (abort or even browser exit)
     requetsSessionId is not passed to tppServer, so redirection is done again
  5 redirectEntity is now only created when redirect comes from tpp. Otherwise not.

### CODE REVIEW

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java [hunk 1]
```diff
@@ -77,16 +65,10 @@ public static void validateSessionCookieValue(String sessionCookieValueString, S
         ObjectMapper mapper = new ObjectMapper();
         SessionCookieValue sessionCookieValue = mapper.readValue(decode, SessionCookieValue.class);
         if (sessionCookieValue.getHashedXsrfToken() == xsrfToken.hashCode()) {
-            log.info("validation of token ok");
+            log.info("validation of token for session ok {}", sessionCookieValue);
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java:68]: While is out of the scope of this PR, hashCode() is not portable across JVMs. Probably using some secure hashes like SHA-2 with low collision probability and portability accross JVMs is better

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/RestRequestContextConfig.java [hunk 2]
```diff
@@ -2,23 +2,35 @@
...
 
+@Slf4j
 @Configuration
 public class RestRequestContextConfig {
     @Bean
     @Scope(scopeName = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
     public RestRequestContext provideCurrentRestRequest(HttpServletRequest httpServletRequest) {
+        String sessionCookieValue = null;
+        if (httpServletRequest.getCookies() != null) {
+            Optional<Cookie> sessionCookie = Arrays.stream(httpServletRequest.getCookies()).filter(cookie -> cookie.getName().equalsIgnoreCase(Consts.COOKIE_SESSION_COOKIE_NAME)).findFirst();
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/RestRequestContextConfig.java:25]: yep

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java (hunk 1 above)
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java:68]: yep, use now sha256

File: fintech-examples/fintech-db-schema/src/main/resources/migration/migrations/0000-init-schema.xml [hunk 3]
```diff
@@ -1,40 +1,26 @@
...
-            <column name="date_from" type="date"/>
-            <column name="date_to" type="date"/>
-            <column name="delta_list" type="BOOLEAN"/>
-            <column name="entry_reference_from" type="VARCHAR(255)"/>
-            <column name="request_action" type="INTEGER">
-                <constraints nullable="false"/>
+                <constraints nullable="false" primaryKey="true" primaryKeyName="opb_login_entity_pkey"/>
             </column>
-            <column name="xsrf_token" type="VARCHAR(255)">
+            <column name="login_time" type="TIMESTAMP without time zone">
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply fintech-examples/fintech-db-schema/src/main/resources/migration/migrations/0000-init-schema.xml:18]: code is generated. I added remark in README.TXT
sed "s#type=\"TIMESTAMP without time zone\"#type=\"TIMESTAMP\"#g" /tmp/0000-init-schema.xml

**Pull request merged**